feat(repeater): adopt self-deploying repeaters

ostridm opened the pull request:
closes #174

derevnjuk commented on src/SecTester.Repeater/IRepeater.cs:
```diff
@@ -6,7 +6,7 @@ namespace SecTester.Repeater;
 public interface IRepeater : IAsyncDisposable
 {
-  string RepeaterId { get; }
+  string? RepeaterId { get; }
```
This doesn't make many sense. Repeater is intended to have ID by the design

derevnjuk commented on src/SecTester.Repeater/IRepeaterFactory.cs:
```diff
@@ -4,5 +4,5 @@ namespace SecTester.Repeater;
 public interface IRepeaterFactory
 {
-  Task<IRepeater> CreateRepeater(RepeaterOptions? options = default);
+  Task<IRepeater> CreateRepeater();
```
The user should be able to pass `NamePrefix`; by default, you use hostname. Just keep the original interface (modify only `RepeaterOptions`)

derevnjuk commented on src/SecTester.Repeater/Bus/SocketIoRepeaterBus.cs:
```diff
@@ -48,13 +49,13 @@ private void DelegateEvents()
     {
       var version = response.GetValue<RepeaterVersion>();
-      UpgradeAvailable?.Invoke(new(version.Version));
+      UpgradeAvailable?.Invoke(new Version(version.Version));
```
Use shorthand option
```suggestion
      UpgradeAvailable?.Invoke(new(version.Version));
```

derevnjuk commented on src/SecTester.Repeater/Bus/SocketIoRepeaterBus.cs:
```diff
@@ -48,13 +49,13 @@ private void DelegateEvents()
     {
       var err = response.GetValue<RepeaterError>();
-      ErrorOccurred?.Invoke(new(err.Message));
+      ErrorOccurred?.Invoke(new Exception(err.Message));
```
ditto
```suggestion
      ErrorOccurred?.Invoke(new(err.Message));
```

derevnjuk commented on src/SecTester.Repeater/Bus/SocketIoRepeaterBus.cs:
```diff
@@ -90,21 +91,27 @@ public async ValueTask DisposeAsync()
 
       _logger.LogDebug("Repeater ({RepeaterId}) deployed", result?.RepeaterId);
+
+      return result?.RepeaterId!;
```
This is not the expected behavior if repeaterId is nil or an empty string. Please reflect this in contract as well

derevnjuk commented on src/SecTester.Repeater/Bus/DefaultRepeaterBusFactory.cs:
```diff
@@ -22,7 +22,7 @@ public DefaultRepeaterBusFactory(Configuration config, ILoggerFactory loggerFact
   }
 
-  public IRepeaterBus Create(string repeaterId)
+  public IRepeaterBus Create(string? hostname = default)
```
```suggestion
  public IRepeaterBus Create(string? namePrefix = default)
```

derevnjuk commented on test/SecTester.Repeater.Tests/SecTester.Repeater.Tests.csproj:
```diff
@@ -10,7 +10,6 @@
   </ItemGroup>
 
   <ItemGroup>
-    <Folder Include="Api"/>
```
Please revert back if folder is still there

ostridm replied:
It does not exist anymore.

dependabot[bot] committed:
build(deps-dev): bump braces from 3.0.2 to 3.0.3

Bumps [braces](https://github.com/micromatch/braces) from 3.0.2 to 3.0.3.
- [Changelog](https://github.com/micromatch/braces/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/braces/compare/3.0.2...3.0.3)

---
updated-dependencies:
- dependency-name: braces
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

ostridm committed:
feat(repeater): adopt self-deploying repeaters

closes #174

ostridm committed:
feat(repeater): adopt self-deploying repeaters

closes #174

ostridm committed:
Merge remote-tracking branch 'origin/dependabot/npm_and_yarn/braces-3.0.3' into feat_#174/adopt-self-delpoying-repeater

ostridm committed:
feat(repeater): fix test

closes #174

ostridm committed:
Merge branch 'master' into feat_#174/adopt-self-delpoying-repeater

derevnjuk commented on src/SecTester.Repeater/Bus/IRepeaterBusFactory.cs:
```diff
@@ -2,5 +2,5 @@ namespace SecTester.Repeater.Bus;
 
 public interface IRepeaterBusFactory
```
See https://github.com/NeuraLegion/sectester-js/pull/197#issuecomment-2168281090. Let's rename to  `IBridgeService`

ostridm committed:
feat(repeater): get rid if same repeater re-deployment

closes #174

ostridm committed:
Merge remote-tracking branch 'origin/feat_#174/adopt-self-delpoying-repeater' into feat_#174/adopt-self-delpoying-repeater

derevnjuk commented:
@ostridm the last commit consists a unwanted chagnes. you have to preserve the ability to use the predefined ID. The user of SecTester.Repeater should be able to run a repeater with a constant ID. Such changes in the public interface must be avoided.

ostridm committed:
revert(repeater): get rid if same repeater re-deployment

This reverts commit 0db132139e41fc89c538665946d4d721326a40a1.
